src/Nethermind/Nethermind.Serialization.Rlp/Rlp.cs

```diff
@@ -412,55 +412,20 @@ public static Rlp Encode(long value)
-                {
-                    if (value < 256L * 256L * 256L * 256L * 256L * 256L)
-                    {
-                        if (value < 256L * 256L * 256L * 256L * 256L)
-                        {
-                            if (value < 256L * 256L * 256L * 256L)
-                            {
-                                if (value < 256 * 256 * 256)
-                                {
-                                    if (value < 256 * 256)
-                                    {
-                                        if (value < 128)
-                                        {
-                                            return new Rlp((byte) value);
-                                        }
-
-                                        return value < 256 ? new Rlp(new byte[] {129, (byte) value}) : new Rlp(new byte[] {130, byte6, (byte) value});
-                                    }
-
-                                    return new Rlp(new byte[] {131, byte5, byte6, (byte) value});
-                                }
-
-                                return new Rlp(new byte[] {132, byte4, byte5, byte6, (byte) value});
-                            }
-
-                            return new Rlp(new byte[] {133, byte3, byte4, byte5, byte6, (byte) value});
-                        }
-
-                        return new Rlp(new byte[] {134, byte2, byte3, byte4, byte5, byte6, (byte) value});
-                    }
-
-                    return new Rlp(new byte[] {135, byte1, byte2, byte3, byte4, byte5, byte6, (byte) value});
-                }
-
-                return new Rlp(new byte[] {136, byte0, byte1, byte2, byte3, byte4, byte5, byte6, (byte) value});
-            }
-
-            return Encode(new BigInteger(value), 8);
+            return value > 0
+                ? value switch
+                {
+                    < 128 => new Rlp((byte)value),
+                    < 256 => new Rlp(new byte[] { 129, (byte)value }),
+                    < 256 * 256 => new Rlp(new byte[] { 130, (byte)(value >> 8), (byte)value }),
+                    < 256 * 256 * 256 => new Rlp(new byte[] { 131, (byte)(value >> 16), (byte)(value >> 8), (byte)value }),
+                    < 256L * 256 * 256 * 256 => new Rlp(new byte[] { 132, (byte)(value >> 24), (byte)(value >> 16), (byte)(value >> 8), (byte)value }),
+                    < 256L * 256 * 256 * 256 * 256 => new Rlp(new byte[] { 133, (byte)(value >> 32), (byte)(value >> 24), (byte)(value >> 16), (byte)(value >> 8), (byte)value }),
+                    < 256L * 256 * 256 * 256 * 256 * 256 => new Rlp(new byte[] { 134, (byte)(value >> 40), (byte)(value >> 32), (byte)(value >> 24), (byte)(value >> 16), (byte)(value >> 8), (byte)value }),
+                    < 256L * 256 * 256 * 256 * 256 * 256 * 256 => new Rlp(new byte[] { 135, (byte)(value >> 48), (byte)(value >> 40), (byte)(value >> 32), (byte)(value >> 24), (byte)(value >> 16), (byte)(value >> 8), (byte)value }),
```

Comment:
I like the change. If it helps improving readability further, we can replace the "matrix" of `256` with hex numbers as follows:
```cs
< 0x80 => new Rlp((byte)value),
< 0x100 => new Rlp(new byte[] { 129, (byte)value }),
< 0x1_0000 => new Rlp(new byte[] { 130, (byte)(value >> 8), (byte)value }),
< 0x100_0000 => new Rlp(new byte[] { 131, (byte)(value >> 16), (byte)(value >> 8), (byte)value }),
< 0x1_0000_0000 => new Rlp(new byte[] { 132, (byte)(value >> 24), (byte)(value >> 16), (byte)(value >> 8), (byte)value }),
< 0x100_0000_0000 => new Rlp(new byte[] { 133, (byte)(value >> 32), (byte)(value >> 24), (byte)(value >> 16), (byte)(value >> 8), (byte)value }),
< 0x1_0000_0000_0000 => new Rlp(new byte[] { 134, (byte)(value >> 40), (byte)(value >> 32), (byte)(value >> 24), (byte)(value >> 16), (byte)(value >> 8), (byte)value }),
< 0x100_0000_0000_0000 => new Rlp(new byte[] { 135, (byte)(value >> 48), (byte)(value >> 40), (byte)(value >> 32), (byte)(value >> 24), (byte)(value >> 16), (byte)(value >> 8), (byte)value }),
```